packages/api/src/external/carequality/process-outbound-patient-discovery-resps.ts

```diff
@@ -62,11 +71,36 @@ export async function processOutboundPatientDiscoveryResps({
       },
     });
 
-    log(`Completed.`);
+    // TODO Pass explicitly
+    const facilityId = cqData?.discoveryFacilityId;
+    const rerunPdOnNewDemographics = cqData?.discoveryRerunPdOnNewDemographics;
+    if (facilityId && rerunPdOnNewDemographics) {
```

Comment:
Wondering what's up w/ verifying facilityId upon running a new PD:
- I get that we need one;
- but, in theory, we wouldn't have started the first PD if we didn't, right?
- so I'm thinking this is an opportunity to improve the type, like, we shouldn't have a CQ "discovery" object without a `facilityId`, should we?

Trying to be more deterministic and minimize the amount of permutations when running and analyzing the code.

Reply:
Agreed -- I had mentioned this in a thread with @Orta21. The discovery object would still be optional on the patient external data type (at least unless we want to overhaul the whole thing) so you'd still have a check, but maybe it would be less confusing.

This is connected to the `TODO Pass explicitly` comments. We don't even need the state for these if we can pass them into this function like `requestId`, `patientId`, etc. Then we just use them without having to fetch the patient and check the patient state.